### DISCUSSION TITLE
How to setup mkdocs preview live server

### DISCUSSION THREAD

**Discussion by jklist** [squidfunk/mkdocs-material#7805] [Q&A] [answered]
### Context

1.docker pull squidfunk/mkdocs-material
2.docker run --rm -it -v /www/wwwroot/Mkdocs:/docs squidfunk/mkdocs-material new .
3.docker run -itd --restart=always --name Mkdocs  -v /www/wwwroot/Mkdocs:/docs  -v /www/wwwroot/Mkdocs/tmp:/tmp -p 8000:8000 --workdir /docs squidfunk/mkdocs-material serve -a 0.0.0.0:8000

I used the above commands to create a website in docker. But I found that every time the container is restarted or stopped and then started again, it creates N folders in the corresponding /www/wwwroot/Mkdocs/tmp. A folder is generated every time it is started. Is there a bug?

### Bug description

![image](https://github.com/user-attachments/assets/6a5b042f-14fd-4e58-892a-6578f65eb6a8)

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
-

### Reproduction

none

### Steps to reproduce

none

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by jklist** [comment]
If you do not add -v /www/wwwroot/Mkdocs/tmp:/tmp, it will be generated in /tmp in the container.

**Comment by niclasheinz** [comment]
The command `mkdocs serve` command automatically recreates the documentation. The temporary build is saved under /tmp.

> MkDocs comes with a built-in dev-server that lets you preview your documentation as you work on it. Make sure you're in the same directory as the mkdocs.yml configuration file, and then start the server by running the mkdocs serve command:
https://www.mkdocs.org/getting-started/

The command `-v /www/wwwroot/Mkdocs/tmp:/tmp` mounts the host directory /www/wwwroot/Mkdocs/tmp to the container's /tmp directory, allowing for shared access to files between the host and the Docker container.

Maybe you are looking for how to build mkdocs using Docker?
  - Building your site: https://squidfunk.github.io/mkdocs-material/creating-your-site/#building-your-site

**Comment by jklist** [comment]
> command 命令会自动重新创建文档。临时版本保存在 /tmp 下。`mkdocs serve`
> 
> > MkDocs 带有一个内置的开发服务器，可让您在处理文档时预览文档。确保您与mkdocs.yml配置文件位于同一目录中，然后通过运行 mkdocs serve 命令启动服务器：https://www.mkdocs.org/getting-started/
> 
> 也许您正在寻找如何使用 Docker 构建 mkdocs？
> 
> * 构建您的网站：https://squidfunk.github.io/mkdocs-material/creating-your-site/#building-your-site


I want to use docker to map the docs folder so that I can update and modify it at any time. How should I do this? Do I need to docker run --rm -it -v ${PWD}:/docs squidfunk/mkdocs-material build after each update?

**Comment by niclasheinz** [comment]
You can simply use 
`docker run --rm -it -p 8000:8000 -v ${PWD}:/docs squidfunk/mkdocs-material`

Docs: [mkdocs-material/creating-your-site/#previewing-as-you-write](https://squidfunk.github.io/mkdocs-material/creating-your-site/#previewing-as-you-write)

Run this command in the root folder of your project where mkdocs.yml is located.

Converting this to a discussion.

**Reply by sheepyoung01** [reply]
> You can simply use `docker run --rm -it -p 8000:8000 -v ${PWD}:/docs squidfunk/mkdocs-material`
> 
> Docs: [mkdocs-material/creating-your-site/#previewing-as-you-write](https://squidfunk.github.io/mkdocs-material/creating-your-site/#previewing-as-you-write)
> 
> Run this command in the root folder of your project where mkdocs.yml is located.
> 
> Converting this to a discussion.

This is also my wanted thing, but I have tried this way and fail to get the effect of real time preview when I just run `docker run --rm -it -p 8000:8000 -v ${PWD}:/docs squidfunk/mkdocs-material` (current path is `/home/young/mkdocs-docker`). 

Specifically, I run this docker command, and change the markdown file `docs/tech/0/test.md` in host dir `/home/young/mkdocs-docker`. And the build operation isn't triggered so that the corresponding page doesn't change. It just execute once while starting the docker container.

And I find that it works while run command `mkdocs serve .`

Live server takes no effect on docker deploy?

<img width="925" height="212" alt="image" src="https://github.com/user-attachments/assets/62ed3252-fa7d-4520-b73a-5fae5f0a2ec5" />

My docker image version of mkdocs is the latest.

**Reply by niclasheinz** [reply]
Hey @sheepyoung01 

Since version 9.6.21, Live Reload no longer works, see #8478.
There is a workaround that may be successful: https://github.com/squidfunk/mkdocs-material/issues/8478#issuecomment-3386761434

**Reply by sheepyoung01** [reply]
> Hey @sheepyoung01
> 
> Since version 9.6.21, Live Reload no longer works, see #8478. There is a workaround that may be successful: [#8478 (comment)](https://github.com/squidfunk/mkdocs-material/issues/8478#issuecomment-3386761434)

Thanks!

**Comment by niclasheinz** [comment, marked as answer]
Hi @jklist 

I have written a [bash script ](https://gist.github.com/niclasheinz/7844e11df16ae904203384d2ce9f912c)that automatically recognises the mkdocs-material docker container and then deletes the /tmp/mkdocs* folders in the docker container itself, except for the very last one.

**Reply by jklist** [reply]
Thank you. Is it built into the latest version?

**Reply by niclasheinz** [reply]
No, it is not integrated in the latest version. However, you can run this manually or via cronjob.

**Comment by MihaelIsaev** [comment]
I've found easy working solution for docker https://github.com/squidfunk/mkdocs-material/issues/8478#issuecomment-3473773434
